bank-sdk/example-app/src/androidTest/java/net/gini/android/bank/sdk/exampleapp/ui/screens/DigitalInvoiceEditButton.kt

```diff
@@ -0,0 +1,167 @@
+        var isElementTitleDisplayed = false
+        onView(withText(resourceId))
+            .check { view, _ ->
+                if (view.isShown()) {
+                    isElementTitleDisplayed = true
+                }
+            }
+        return isElementTitleDisplayed
+    }
+
+    fun editElementTextOnArticleBottomSheet(resourceId: Int, text: String) {
+        onView((withId(resourceId))).perform(click()).perform(replaceText(text))
+    }
+
+    fun removeTextFromElementOnArticleBottomSheet(resourceId: Int) {
+        onView((withId(resourceId))).perform(click()).perform(clearText())
+    }
+
+    fun verifyInlineErrorOnElement(resourceId: Int, inlineError: String): Boolean {
+        var isInlineErrorDisplayed = false
+        onView(allOf(withId(resourceId),
+            withText(inlineError)))
+            .check { view, _ ->
+            if (view.isShown()) {
+                isInlineErrorDisplayed = true
+            }
+        }
+            return isInlineErrorDisplayed
+    }
+
+    fun clickSaveButtonOnEditArticleBottomSheet() {
+        onView(withText(net.gini.android.bank.sdk.R.string.gbs_digital_invoice_line_item_details_save)).perform(click())
+    }
+
+    fun clickCancelButtonOnEditArticleBottomSheet() {
+        onView(withId(net.gini.android.bank.sdk.R.id.gbs_close_bottom_sheet)).perform(click())
+    }
+
+    fun saveNameValueOnDigitalInvoiceScreen() {
+        val uiCollection =
+            UiCollection(UiSelector().className("android.view.ViewGroup"))
+        val textView = uiCollection.getChildByInstance(
+            UiSelector().className("android.widget.TextView")
+                .resourceId("R.id.gbs_description")
+            , 0)
+        if(textView.exists() && textView.isEnabled) {
+            textView.text
+        }
+    }
+
+    fun checkNameIsUpdated(text: String): Boolean {
+        val uiCollection =
+            UiCollection(UiSelector().className("android.view.ViewGroup"))
+        val textView = uiCollection.getChildByInstance(
+            UiSelector().className("android.widget.TextView")
+                .resourceId("net.gini.android.bank.sdk.exampleapp:id/gbs_description")
+            , 0)
+        if (textView.exists()) {
+            val actualText = textView.text.replace(Regex("\\d+x\\s*"), "")
+            if (actualText == text) {
+                return true
+            }
+        }
+        return false
+    }
+
+    fun checkUnitPriceIsUpdated(text: String): Boolean {
+        val uiCollection =
+            UiCollection(UiSelector().className("android.view.ViewGroup"))
+        val textView = uiCollection.getChildByInstance(
+            UiSelector().className("android.widget.TextView")
+                .resourceId("net.gini.android.bank.sdk.exampleapp:id/gbs_per_unit")
+            , 0)
+        if (textView.exists()) {
+            val actualText = textView.text
+            val expectedText = "€$text per unit"
```

Comment:
we can use this `gbs_digital_invoice_line_item_quantity` in strings.xml

Reply:
can you help me explain where to change it?

Reply:
We tried it over our call and unable to do it because Espresso doesn't have the access to the resources.